src/main/java/org/dcsa/ebl/service/impl/TransportDocumentTOServiceImpl.java

```diff
@@ -1,28 +1,207 @@
+                                                                }
+
+                                                                if (carrierBookingReference == null) {
+                                                                    return Mono.error(new GetException("No CarrierBookingReference specified on ShippingInstruction:" + shippingInstruction.getId() + " - internal error!"));
+                                                                } else {
+//                                                                    return Flux.concat(
+                                                                            return bookingService.findById(carrierBookingReference)
+                                                                                    .flatMap(booking -> {
+                                                                                        transportDocumentTO.setServiceTypeAtOrigin(booking.getServiceTypeAtOrigin());
+                                                                                        transportDocumentTO.setServiceTypeAtDestination(booking.getServiceTypeAtDestination());
+                                                                                        transportDocumentTO.setShipmentTermAtOrigin(booking.getShipmentTermAtOrigin());
+                                                                                        transportDocumentTO.setShipmentTermAtDestination(booking.getShipmentTermAtDestination());
+                                                                                        transportDocumentTO.setServiceContract(booking.getServiceContract());
+                                                                                        return Mono.just(booking);
+                                                                                    })
+//                                                                            ,shipmentService.findByCarrierBookingReference(carrierBookingReference)
+//                                                                                    .next()
+//                                                                                    .flatMap(shipment -> {
+//                                                                                        shipmentTransportService.findByShipmentIDOrderBySequenceNumber(shipment.getId())
+//                                                                                                .next()
+//                                                                                                .flatMap(shipmentTransport -> xxx)
+//                                                                                    })
+//                                                                    )
+//                                                                    .count()
+                                                                    .thenReturn(shippingInstruction);
```

Comment:
I am not sure where this return value is needed?  (It might be useful; it is hard to tell in the github UI)